src/metaxy/cli/metadata.py

```diff
@@ -19,6 +21,265 @@
+            except ValueError as e:
+                if format == "json":
+                    error_data = {"error": "Snapshot error", "message": str(e)}
+                    print(json.dumps(error_data))
+                else:
+                    console.print(f"[red]✗[/red] {e}")
+                raise SystemExit(1)
+            except ImportError as e:
+                if format == "json":
+                    error_data = {"error": "Failed to load snapshot", "message": str(e)}
+                    print(json.dumps(error_data))
+                else:
+                    console.print(f"[red]✗[/red] Failed to load snapshot: {e}")
+                    console.print(
+                        "[yellow]Hint:[/yellow] Feature classes may have been moved or deleted."
+                    )
+                raise SystemExit(1)
+        else:
+            # Use current graph from context
+            graph = context.graph
+
+        # Verify all requested features exist in the graph
+        missing_in_graph = [
+            key for key in parsed_keys if key not in graph.features_by_key
+        ]
+        if missing_in_graph:
+            formatted = ", ".join(key.to_string() for key in missing_in_graph)
+            if assert_in_sync:
+                # Only exit if assert_in_sync is enabled
+                if format == "json":
+                    error_data = {
+                        "error": "Features not found in graph",
+                        "features": [k.to_string() for k in missing_in_graph],
+                    }
+                    print(json.dumps(error_data))
+                else:
+                    console.print(
+                        f"[red]✗[/red] Feature(s) not found in graph: {formatted}"
+                    )
+                raise SystemExit(1)
+            else:
+                # Otherwise, warn and continue with what we have
+                if format == "text":
+                    console.print(
+                        f"[yellow]Warning:[/yellow] Feature(s) not found in graph: {formatted}"
+                    )
+                # Filter to only existing features
+                parsed_keys = [
+                    key for key in parsed_keys if key in graph.features_by_key
+                ]
+                if not parsed_keys:
+                    if format == "json":
+                        print(
+                            json.dumps(
+                                {
+                                    "features": [],
+                                    "warning": "No valid features to check",
+                                }
+                            )
+                        )
+                    else:
+                        console.print(
+                            "[yellow]Warning:[/yellow] No valid features to check."
+                        )
+                    return
+
+        needs_update = False
+
+        # Print header for text format
+        if format == "text":
```

Comment:
it's weird to do this check before json printing branch

let's do just one format check, not two